keccak256/src/keccak_arith.rs

```diff
@@ -127,7 +126,10 @@ impl Default for Keccak {
 
 impl Keccak {
     pub fn update(&mut self, input: &[u8]) {
+        println!("State just after call update {:?}", self.state);
+        println!("input just after call update {:?}", input);
         let padding_total = self.sponge.rate - (input.len() % self.sponge.rate);
+        println!("Padding total {:?}", padding_total);
```

Comment:
Do we accidentally add these?

```suggestion
        let padding_total = self.sponge.rate - (input.len() % self.sponge.rate);
```